src/main/java/net/minecraftforge/registries/RegistryObject.java

```diff
@@ -42,7 +62,7 @@ public final class RegistryObject<T> implements Supplier<T>
      */
     public static <T, U extends T> RegistryObject<U> create(final ResourceLocation name, IForgeRegistry<T> registry)
     {
-        return new RegistryObject<U>(name, registry);
+        return create(name, registry.getRegistryKey(), "");
```

Comment:
Instead of an empty string maybe we should do something like `"unknown"` or `"unknown mod"`, given it is only really used for logging and I think then provides better logging information.

Reply:
Objections to `ModLoadingContext.get().getActiveNamespace()`?  This ctor is used by defreg normally

Reply:
Sounds like a reasonable solution